Apps/Admin/Server/Services/SupportService.cs

```diff
@@ -203,33 +179,32 @@ public async Task BlockAccessAsync(string hdid, IEnumerable<DataSource> dataSour
             await this.patientRepository.BlockAccessAsync(blockAccessCommand, ct);
         }
 
-        private async Task<IList<PatientSupportDependentInfo>> GetDependentsAsync(string delegateHdid, CancellationToken ct)
+        private async Task<PatientModel> GetPatientAsync(PatientDetailsQuery query, CancellationToken ct = default)
+        {
+            PatientModel? patient = (await this.patientRepository.Query(query, ct)).Items.SingleOrDefault();
+            return patient ?? throw new ProblemDetailsException(ExceptionUtility.CreateProblemDetails(ErrorMessages.ClientRegistryRecordsNotFound, HttpStatusCode.NotFound, nameof(SupportService)));
+        }
+
+        private async Task<PatientModel?> TryGetPatientAsync(PatientIdentifierType identifierType, string queryString, CancellationToken ct)
```

Comment:
I have not better suggestion but with the `Try` prefix, it feels like it should be a predicate with an `out` parameter.

Reply:
Yeah, I couldn't think of anything better. GetPatientOrNullAsync was my number 2 choice.

Reply:
Hmm the second one is too on the nose. I'm happy to adjust my perspective for the sake of the nullable type